e2e/tests/test_snapp_deposit_withdraw.rs

```diff
@@ -0,0 +1,152 @@
+use e2e::common::{wait_for, wait_for_condition, FutureWaitExt};
+use e2e::snapp::{await_state_transition, setup_snapp};
+use ethcontract::web3::api::Web3;
+use ethcontract::web3::transports::Http;
+use ethcontract::web3::types::{H160, H256, U128, U256};
+use ethcontract::Account;
+use std::str::FromStr;
+
+#[test]
+fn test_deposit_and_withdraw() {
+    let (eloop, http) = Http::new("http://localhost:8545").expect("transport failed");
+    eloop.into_remote();
+    let web3 = Web3::new(http);
+    let (instance, accounts, tokens, db) = setup_snapp(&web3, 3, 3);
+
+    // Test environment values
+    let deposit_amount = 18_000_000_000_000_000_000u128;
+
+    let user_address = &accounts[2];
```

Comment:
nit: `&` is probably unnecessary here since `H160` implements `Copy`. This will also make it so you don't have to `*user_address` everywhere, and can just use `user_address`.

Reply:
I originally tried using copy here, but "it" wouldn't allow me to... Actually turns out my IDE has been lying to me!